ccmlib/common.py

```diff
@@ -748,12 +748,12 @@ def pid_listening_on(addr: str, port: int):
                                  f'src = inet:{addr}:{port}'],
                                 stdout=subprocess.PIPE,
                                 check=True)
-        matched = re.search(r'pid=(\d+)', result.stdout)
+        matched = re.search(r'pid=(\d+)', str(result.stdout))
         if matched is not None:
-            pid = int(matched.group(1))
+            pids = [int(matched.group(1))]
```

Comment:
if we are interested in find all matches, probably we can use 
```py
matches = re.findall(r'pid=(\d+)', result.stdout)
pids = [int(match) for match in matches]
```

because, `re.search()` only returns the first match. FWIW, there is no need to pass `re.MULTILINE` as the `flags` passed to `re.findall()`.

Reply:
yeah, I thought about it

but in the situation I go, this command didn't really returned multiple processes, while `lsof` did
so I gave it up.

but can change that one as well

Reply:
replaced search with findall